src/components/taskDetails/index.tsx

```diff
@@ -0,0 +1,228 @@
+import React, {
+  ChangeEvent,
+  FC,
+  useEffect,
+  useContext,
+  useReducer,
+  useRef,
+  useState,
+} from 'react';
+import useFetch from '@/hooks/useFetch';
+import { isUserAuthorizedContext } from '@/context/isUserAuthorized';
+import NavBar from '@/components/navBar/index';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import updateTaskDetails from '@/helperFunctions/updateTaskDetails';
+import task from '@/interfaces/task.type';
+import classNames from './task-details.module.scss';
+import TaskContainer from './TaskContainer';
+import Details from './Details';
+type Props = {
+  url: string;
+  taskID: string;
+};
+
+const initialState = {
+  taskDetails: {} as task,
+  editedDetails: {} as task,
+};
+
+function reducer(state: any, action: any) {
+  switch (action.type) {
+    case 'setTaskDetails':
+      return {
+        ...state,
+        taskDetails: { ...state.taskDetails, ...action.payload },
+      };
+    case 'setEditedDetails':
+      return {
+        ...state,
+        editedDetails: { ...state.editedDetails, ...action.payload },
+      };
+    case 'reset':
+      return { ...state, taskDetails: { ...action.payload } };
+    default:
+      return state;
+  }
+}
+
+const TaskDetails: FC<Props> = ({ url, taskID }) => {
+  const isAuthorized = useContext(isUserAuthorizedContext);
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+  const initialDataRef = useRef<task>();
+  const { response, error, isLoading } = useFetch(url);
+  const { SUCCESS, ERROR } = ToastTypes;
+  const { taskDetails } = state;
+  useEffect(() => {
+    const fetchedData: task = { ...response.taskData };
+    dispatch({ type: 'setTaskDetails', payload: fetchedData });
+    initialDataRef.current = fetchedData;
+  }, [isLoading, response]);
```

Comment:
NIT: you can use rtk query and avoid using useEffect and useFetch and useReducer

Reply:
Will update it in the next version of this feature 👍🏻